### ISSUE TITLE
Does not parse accentuated characters in linted code

### ISSUE THREAD

**Issue by ClementPhan** [microsoft/vscode-pylint#23]
# Description
When a file contains accentuated characters ("é" for example), the extension is not able to parse the source code.

# Steps to reproduce

1. Create file ```stuff.py``` with contents: 
```
a = "é"

```
2. Save 

# Expected output

Taken from the python extension with  as seen in "Output: Python" with following settings :
```
"python.linting.pylintEnabled": true,
"python.linting.ignorePatterns": [
  "**/site-packages/**/*.py",
  ".vscode/*.py",
  "dist/*",
  ".venv/*"
]
```
```
##########Linting Output - pylint##########

[
    {
        "type": "convention",
        "module": "stuff",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "stuff.py",
        "symbol": "missing-module-docstring",
        "message": "Missing module docstring",
        "message-id": "C0114"
    },
    {
        "type": "convention",
        "module": "stuff",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": 1,
        "endColumn": 1,
        "path": "stuff.py",
        "symbol": "invalid-name",
        "message": "Constant name \"a\" doesn't conform to UPPER_CASE naming style",
        "message-id": "C0103"
    }
]
```
# Actual output
Taken from Output: Pylint
```
Failed to handle notification "textDocument/didSave": text_document=TextDocumentIdentifier(uri='file:///c%3A/Users/phancl01/OneDrive%20-%20SOMFY/Desktop/temp/tests/stuff.py') text=None
Traceback (most recent call last):
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pygls\protocol.py", line 316, in _handle_notification
    self._execute_notification(handler, params)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pygls\protocol.py", line 223, in _execute_notification
    handler(*params)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\linter\linter_server.py", line 173, in did_save
    _lint_and_publish_diagnostics(params)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\linter\linter_server.py", line 130, in _lint_and_publish_diagnostics
    result = utils.run_module(module, argv, use_stdin, document.source)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\linter\utils.py", line 194, in run_module
    runpy.run_module(module, run_name="__main__")
  File "C:\Users\phancl01\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 212, in run_module
    return _run_code(code, {}, init_globals, run_name, mod_spec)
  File "C:\Users\phancl01\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pylint\__main__.py", line 9, in <module>
    pylint.run_pylint()
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pylint\__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pylint\lint\run.py", line 398, in __init__
    linter.check(args)
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pylint\lint\pylinter.py", line 991, in check
    functools.partial(self.get_ast, data=_read_stdin()),
  File "c:\Users\phancl01\.vscode\extensions\ms-python.pylint-2022.1.10811002\bundled\libs\pylint\lint\pylinter.py", line 68, in _read_stdin
    return sys.stdin.read()
  File "C:\Users\phancl01\AppData\Local\Programs\Python\Python310\lib\codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 5: invalid continuation byte
```

# Environment 
Windows 10 20H2
Visual studio code 1.66.0
Python extension v2022.4.0
Pylint extension v2022.1.10811002

**Comment by karthiknadig** [comment]
@ClementPhan What is the encoding on the file? this is so I can test it locally.

**Comment by ClementPhan** [comment]
It should be utf8

**Comment by brettcannon** [comment]
Are you able to provide the code that's failing? Or at least an example file that triggers the same problem?

**Comment by karthiknadig** [comment]
@ClementPhan I am not able to re-produce this with utf-8 file encoding. since this is windows, it can sometimes get saved as the Windows default code page.

**Comment by brettcannon** [comment]
And can you also show what Pylint shows when run from the terminal?

**Comment by ClementPhan** [comment]
![image](https://user-images.githubusercontent.com/9255146/161114664-16414d48-d975-4744-b0b9-7a1a1b822fbd.png)

This is my setup, with vscode telling me i am in utf8.

The pylint output is seen in the image, but it is installed from pip.

The file is attached as a txt : [temp.txt](https://github.com/microsoft/vscode-pylint/files/8391809/temp.txt)

**Comment by ClementPhan** [comment]
> @ClementPhan I am not able to re-produce this with utf-8 file encoding. since this is windows, it can sometimes get saved as the Windows default code page.

Can you tell me how to check for this ?

**Comment by karthiknadig** [comment]
> Can you tell me how to check for this ?

@ClementPhan I don't think there is an easy way to check that. But given that VS Code is saying it is UTF-8 I think we can assume that for now. 

Can you share what you have for the environment variables and the default encoding for your file system?
for the default FS encoding you can run this:
```py
import sys
print(f"FileSystem: {sys.getfilesystemencoding()}")
print(f"Default: {sys.getdefaultencoding()}")
```

**Comment by brettcannon** [comment]
It looks like Pylint forces encoding to UTF-8 on stdin:

https://github.com/PyCQA/pylint/blame/f46904ab9b058d8506b87cedd1f0c710d5c19360/pylint/lint/pylinter.py#L82-L85

And that code is at least 2 years old, so this isn't a new thing. So we may have to forcibly use UTF-8 when working with Pylint.

**Comment by ClementPhan** [comment]
Here is everything.

<details><summary>Environment variables:</summary>
<p>

```
ALLUSERSPROFILE C:\ProgramData
APPDATA C:\Users\xxx\AppData\Roaming
AWP_LOCALE212 en-us
COMMONPROGRAMFILES C:\Program Files\Common Files
COMMONPROGRAMFILES(X86) C:\Program Files (x86)\Common Files
COMMONPROGRAMW6432 C:\Program Files\Common Files
COMPUTERNAME S7Y9J573
COMSPEC C:\WINDOWS\system32\cmd.exe
DRIVERDATA C:\Windows\System32\Drivers\DriverData
FPLOCNAME en-US
HOMEDRIVE C:
HOMEPATH \Users\xxx
LOCALAPPDATA C:\Users\xxx\AppData\Local
LOGONSERVER \\SRW0005P
NEWENVIRONMENT2 FPLCID
NUMBER_OF_PROCESSORS 12
ONEDRIVE C:\Users\xxx\OneDrive - SOMFY
ONEDRIVECOMMERCIAL C:\Users\xxx\OneDrive - SOMFY
OS Windows_NT
PATH C:\Users\xxx\source\repos\frein3\.venv\Scripts;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Users\xxx\AppData\Local\Programs\Python\Python310\Scripts\;C:\Users\xxx\AppData\Local\Programs\Python\Python310\;C:\Users\xxx\AppData\Local\Microsoft\WindowsApps;C:\Users\xxx\AppData\Local\Programs\Microsoft VS Code\bin
PATHEXT .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL
PROCESSOR_ARCHITECTURE AMD64
PROCESSOR_IDENTIFIER Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
PROCESSOR_LEVEL 6
PROCESSOR_REVISION a502
PROE_START_CMD212 C:\Program Files\PTC\Creo 7.0.4.0\Parametric\bin\parametric.bat
PROGRAMDATA C:\ProgramData
PROGRAMFILES C:\Program Files
PROGRAMFILES(X86) C:\Program Files (x86)
PROGRAMW6432 C:\Program Files
PSMODULEPATH C:\Users\xxx\OneDrive - SOMFY\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Microsoft Azure Information Protection\Powershell
PUBLIC C:\Users\Public
SYSTEMDRIVE C:
SYSTEMROOT C:\WINDOWS
TC_SEV_ENV PRODUCTION
TEMP C:\Users\xxx\AppData\Local\Temp
TMP C:\Users\xxx\AppData\Local\Temp
UATDATA C:\WINDOWS\CCM\UATData\D9F8C395-CAB8-491d-B8AC-179A1FE1BE77
USERDNSDOMAIN XXX
USERDOMAIN XXX
USERDOMAIN_ROAMINGPROFILE XXX
USERNAME xxx
USERPROFILE C:\Users\xxx
VIRTUAL_ENV C:\Users\xxx\source\repos\frein3\.venv
VIRTUAL_ENV_PROMPT .venv
WINDIR C:\WINDOWS
WSLENV WT_SESSION::WT_PROFILE_ID
WT_PROFILE_ID {61c54bbd-c2c6-5271-96e7-009a87ff44bf}
WT_SESSION 15acbf6d-3764-4734-8492-4de3e7c8d31d
ZES_ENABLE_SYSMAN 1
_OLD_VIRTUAL_PATH C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Users\xxx\AppData\Local\Programs\Python\Python310\Scripts\;C:\Users\xxx\AppData\Local\Programs\Python\Python310\;C:\Users\xxx\AppData\Local\Microsoft\WindowsApps;C:\Users\xxx\AppData\Local\Programs\Microsoft VS Code\bin
```
</p>
</details>

<details><summary>File encoding:</summary>
<p>

```python
>>> import sys
>>> print(f"FileSystem: {sys.getfilesystemencoding()}")
FileSystem: utf-8
>>> print(f"Default: {sys.getdefaultencoding()}")
Default: utf-8
```
</p>
</details>

**Comment by karthiknadig** [comment]
@ClementPhan The latest pre-release should handle this better, we have changed the encoding to be always utf-8. Please do try with the latest.

**Comment by ClementPhan** [comment]
I removed my mitigation and it seems to works properly for me.

Thank you very much.
